File: Phase01/src/InputProcessor.java
import java.util.ArrayList;

public class InputProcessor {
    private final ArrayList<String> necessary = new ArrayList<>();
    private final ArrayList<String> optional = new ArrayList<>();
    private final ArrayList<String> avoided = new ArrayList<>();

    public ArrayList<String> getNecessary() {
        return necessary;
    }

    public ArrayList<String> getOptional() {
        return optional;
    }

    public ArrayList<String> getAvoided() {
        return avoided;
    }

    public void process(String[] input) {
        for (String s : input) {
            if (s.startsWith("+")) optional.add(s.substring(1));
            else if (s.startsWith("-")) avoided.add(s.substring(1));
            else necessary.add(s);
        }
    }
}

File: Phase05/SampleLibrary/SampleProject/InputProcessor.cs
using System.Collections;

namespace SampleProject;

public class InputProcessor
{
    public HashSet<string> necessary = new();
    public HashSet<string> optional = new();
    public HashSet<string> avoided = new();

    public void Process(String[] input) {
        foreach (string s in input) {
            if (s.StartsWith("+")) optional.Add(s.Substring(1));
            else if (s.StartsWith("-")) avoided.Add(s.Substring(1));
            else necessary.Add(s);
        }
    }
}